src/Authenticators/LDAPChangePasswordHandler.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace SilverStripe\ActiveDirectory\Authenticators;
+
+use Exception;
+use SilverStripe\ActiveDirectory\Forms\LDAPChangePasswordForm;
+use SilverStripe\ActiveDirectory\Services\LDAPService;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTP;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\ORM\ValidationResult;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\MemberAuthenticator\ChangePasswordHandler;
+use SilverStripe\Security\Security;
+
+class LDAPChangePasswordHandler extends ChangePasswordHandler
+{
+    /**
+     * @var array Allowed Actions
+     */
+    private static $allowed_actions = [
+        'changepassword',
+        'changePasswordForm',
+    ];
+
+    /**
+     * Factory method for the lost password form
+     *
+     * @return LDAPChangePasswordForm Returns the lost password form
+     */
+    public function changePasswordForm()
+    {
+        return LDAPChangePasswordForm::create($this, 'ChangePasswordForm');
+    }
+
+    /**
+     * Change the password
+     *
+     * @param array $data The user submitted data
+     * @param LDAPChangePasswordForm $form
+     * @return HTTPResponse
+     */
+    public function doChangePassword(array $data, $form)
+    {
+        /**
+         * @var LDAPService $service
+         */
+        $service = Injector::inst()->get(LDAPService::class);
+        $member = Security::getCurrentUser();
+        if ($member) {
+            try {
+                $userData = $service->getUserByGUID($member->GUID);
+            } catch (Exception $e) {
+                Injector::inst()->get('Logger')->error($e->getMessage());
```

Comment:
This changed a while back, it's now `Injector::inst()->get(LoggerInterface::class)` with `use Psr\Log\LoggerInterface`

Reply:
Aah yes. Thanks.